application/src/main/java/org/togetherjava/tjbot/features/help/PinnedAnnouncementListener.java

```diff
@@ -0,0 +1,56 @@
+package org.togetherjava.tjbot.features.help;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageType;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.entities.channel.unions.MessageChannelUnion;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.features.MessageReceiverAdapter;
+
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * Listener that receives all pinned announcement messages from helper forum so that it can be
+ * removed
+ */
+public final class PinnedAnnouncementListener extends MessageReceiverAdapter {
+
+    private final Predicate<String> isHelpForumName;
+
+    /**
+     * Creates a new listener to receive all pinned announcement messages sent in help channels.
+     *
+     * @param config the config to use for this
+     */
+    public PinnedAnnouncementListener(Config config) {
+        isHelpForumName =
+                Pattern.compile(config.getHelpSystem().getHelpForumPattern()).asMatchPredicate();
+    }
+
+    @Override
+    public void onMessageReceived(MessageReceivedEvent event) {
+        removePinnedAnnouncement(event);
+    }
```

Comment:
theres a problem with naming and responsibilities here. a method called `removePinnedAnnocuncement` sounds unconditional. but ur method still checks if its the right thread.

id suggest to either move the `if` stuff to this method, but then the question is if an extra method is still justified.
or to rename it to `handlePinnedAnnouncement`, to make it clear that it didnt decide anything yet

Reply:
Agreed, you have a solid reasoning.